lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,187 @@
+# todo those ought to be in the gemspec
+require 'tempfile'
+require 'crowdin-api'
+require 'yaml'
+require 'zip'
+require 'pathname'
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+
+class LocalesUpdater
+  extend TmpDirectory
+
+  def self.update_all_locales_of_all_repos(debug: true)
+    repos_to_update = plugins_with_locales
+
+    tmp_path = create_tmp_directory(delete: true)
+
+    Dir.chdir tmp_path do
+      repos_to_update.each do |name, specifics|
+        # todo each branch
+        uri = specifics[:uri]
+        set_crowdin_specifics(specifics)
```

Comment:
I think you should redo what you did with `GitRepository` and apply that same abstraction to Crowdin so that if we ever switch to a different service, we only have to make changes there.